src/plan/marksweep/gc_work.rs

```diff
@@ -0,0 +1,56 @@
+use crate::plan::global::NoCopy;
+use crate::plan::global::Plan;
+use crate::policy::space::Space;
+use crate::scheduler::gc_work::*;
+use crate::util::Address;
+use crate::util::ObjectReference;
+use crate::vm::VMBinding;
+use crate::MMTK;
+use std::ops::{Deref, DerefMut};
+
+use super::MarkSweep;
+
+pub struct MSProcessEdges<VM: VMBinding> {
+    plan: &'static MarkSweep<VM>,
+    base: ProcessEdgesBase<MSProcessEdges<VM>>,
+}
+
+impl<VM: VMBinding> ProcessEdgesWork for MSProcessEdges<VM> {
+    type VM = VM;
+    const OVERWRITE_REFERENCE: bool = false;
+    fn new(edges: Vec<Address>, _roots: bool, mmtk: &'static MMTK<VM>) -> Self {
+        let base = ProcessEdgesBase::new(edges, mmtk);
+        let plan = base.plan().downcast_ref::<MarkSweep<VM>>().unwrap();
+        Self { plan, base }
+    }
+
+    #[inline]
+    fn trace_object(&mut self, object: ObjectReference) -> ObjectReference {
+        if object.is_null() {
+            return object;
+        }
+        trace!("Tracing object {}", object);
+        if self.plan.ms_space().in_space(object) {
```

Comment:
Not relevant to this PR, but I think the `in_space(...)` should be renamed to `contains_object(...)` or simply `contains(...)`, and `address_in_space(...)` should be renamed to `contains_address(...)`.

Reply:
Yeah. Good point. Maybe I will do this renaming separately.